src/contracts/services/shadeStaking.ts

```diff
@@ -0,0 +1,93 @@
+import { getActiveQueryClient$ } from '~/client';
+import { sendSecretClientContractQuery$ } from '~/client/services/clientServices';
+import {
+  switchMap,
+  first,
+  map,
+  lastValueFrom,
+} from 'rxjs';
+import { convertCoinFromUDenom } from '~/lib/utils';
+import { msgQueryShadeStakingOpportunity } from '~/contracts/definitions/shadeStaking';
+import {
+  StakingInfoServiceResponse,
+  StakingRewardPoolServiceModel,
+  StakingInfoServiceModel,
+} from '~/types/contracts/shadeStaking/index';
+
+/**
+ * parses the response from the shade staking contract into a model
+ */
+function parseStakingOpportunity(data: StakingInfoServiceResponse): StakingInfoServiceModel {
+  const stakeTokenAddress = data.staking_info.info.stake_token;
+  const totalStakedRaw = data.staking_info.info.total_staked;
+  const unbondingPeriod = Number(data.staking_info.info.unbond_period);
+  const rewardPools: StakingRewardPoolServiceModel[] = data.staking_info.info.reward_pools
+    .map((reward) => ({
+      id: reward.id,
+      amountRaw: reward.amount,
+      startDate: new Date(Number(reward.start) * 1000),
+      endDate: new Date(Number(reward.end) * 1000),
+      tokenAddress: reward.token.address,
+      // data returned from the contract in normalized form with
+      // 18 decimals, in addition to any decimals on the individual token
+      rateRaw: convertCoinFromUDenom(reward.rate, 18).toString(),
```

Comment:
Instead of hardcoding 18 here, let's add a NOMALIZATION_FACTOR const at the top of the file to be referenced.